snakemake/logging.py

```
@@ -305,145 +244,365 @@
-                    datetime.datetime.now().isoformat().replace(":", "")
-                    + ".snakemake.log",
-                )
-            )
 
-            self.logfile_handler = _logging.FileHandler(self.logfile)
-            self.logger.addHandler(self.logfile_handler)
+def show_logs(logs):
+    """Helper method to show logs."""
+    for f in logs:
+        try:
+            content = open(f, "r").read()
```

Ruff

Use a context manager for opening files

(SIM115)